app/assets/javascripts/app/views/notifications_view.js

```diff
@@ -49,47 +41,57 @@ app.views.Notifications = Backbone.View.extend({
     this.updateView(data["guid"], type, data["unread"]);
   },
 
+  /* TODO: Test this */
+  markAllRead: function(){
+    var self = this;
+    this.getAllUnread().each(function(i, el){
+      self.setRead($(el).data("guid"));
+    });
+  },
+
   updateView: function(guid, type, unread) {
     var change = unread ? 1 : -1,
         all_notes = $('ul.nav > li:eq(0) .badge'),
         type_notes = $('ul.nav > li[data-type=' + type + '] .badge'),
         header_badge = $('#notification_badge .badge_count'),
-        note = $('.stream_element[data-guid=' + guid + ']');
+        note = $('.stream_element[data-guid=' + guid + ']'),
+        markAllRead = $('a#mark_all_read_link'),
+        translationKey = 'read';
```

Comment:
Before it was `notifications.mark_read`, now it is just `read`. I don't think that will work. Btw you could also write
`translationKey = unread ? 'notifications.mark_read' : 'notifications.mark_unread'`